kart/working_copy/base.py

```diff
@@ -694,6 +695,21 @@ def diff_db_to_tree_feature(
 
         return feature_diff
 
+    def _fetch_if_promised(self, feature_error, dataset, feature_pk):
+        subcode = getattr(feature_error, 'subcode', 0)
+
+        if subcode == LibgitSubcode.ENOSUCHPATH:
+            # There is no such feature. This is okay: it just means the user has inserted
+            # the feature into the working copy and it has not yet been committed.
+            return None
+        elif subcode == LibgitSubcode.EOBJECTPROMISED:
+            # There is such a feature, but we don't have it locally right now. Fetch it.
+            dataset.fetch_missing_dirty_features(self)
+            return dataset.get_feature(feature_pk)
+        else:
+            # Some other error has happened, or no subcode was found. Re-raise the error.
+            raise feature_error
```

Comment:
This will cause the traceback to show `_fetch_if_promised` rather than `dataset.get_feature`, which might be confusing. Just use bare `raise` here to preserve the original traceback.

However, since it's confusing for a function to use bare `raise` when it doesn't appear to ahve a try/except, move that into the function too, and call it `get_feature_and_fetch_if_needed` or something.

Reply:
Done